### DISCUSSION TITLE
`MediaQuery` integration

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#202] [Q&A]
Hi, [here](https://api.flutter.dev/flutter/widgets/MediaQuery/of.html) it says:

```
When any of that information changes, your widget will be scheduled to be rebuilt, keeping your widget up-to-date.
```

Is there an obvious ReArch integration here? I just have done a Side Effect but I have two worries:
1. That the scheduled rebuilds break ReArch flow.
2. That my side effect will not update on scheduled rebuilds.

```dart
MediaQueryData mediaQuery(BuildContext context) =>
      use.callonce(() => MediaQuery.of(context));
```

**Comment by busslina** [comment]
My bad, it works completely fine !

`RearchConsumer.build()`:
```dart
// Media query
    final mediaQuery = MediaQuery.of(context);
    final screenSize = mediaQuery.size;

    print('SIZE: $screenSize');
```

Output:
```
SIZE: Size(588.0, 566.0)
SIZE: Size(589.3, 566.0)
SIZE: Size(654.0, 566.0)
SIZE: Size(657.3, 566.0)
SIZE: Size(824.0, 566.0)
SIZE: Size(950.0, 566.0)
SIZE: Size(952.0, 566.0)
SIZE: Size(952.0, 567.3)
SIZE: Size(952.0, 583.3)
SIZE: Size(952.0, 633.3)
SIZE: Size(952.0, 641.3)
SIZE: Size(1164.0, 646.0)
```

**Comment by GregoryConrad** [comment]
ReArch works fine with InheritedWidgets, like MediaQuery.

I often make accessing MediaQuery an extension on BuildContext, but you can also make a side effect for it too if you wanted:

```dart
extension UseMediaQuery on WidgetSideEffectRegistrar {
  MediaQuery mediaQuery() => MediaQuery.of(context());
}

// …
use.mediaQuery();
```

**Reply by busslina** [reply]
Yeah, I though about it and I think it is better not to put "side effects" that doesn't use "use".

**Reply by GregoryConrad** [reply]
Oh, it does use `use` for the `context()`. I just didn't write out the convenience thing so that you can `use.` in new effects (since I was writing on my phone).

A more fully typed out example:

```dart
extension UseMediaQuery on WidgetSideEffectRegistrar {
  WidgetSideEffectRegistrar use => this;
  MediaQuery mediaQuery() => MediaQuery.of(use.context());
}
```

**Reply by busslina** [reply]
Cool !!
